strawberryfields/backends/states.py

```diff
@@ -359,6 +362,56 @@ def quad_expectation(self, mode, phi=0, **kwargs):
         """
         raise NotImplementedError
 
+    @abc.abstractmethod
+    def poly_quad_expectation(self, A, d=None, k=0, phi=0, **kwargs):
+        r"""The multi-mode expectation values and variance of arbitrary 2nd order polynomials
+        of quadrature operators.
+
+        An arbitrary 2nd order polynomial of quadrature operators over $N$ modes can always
+        be written in the following form:
+
+        .. math:: P(\mathbf{r}) = \frac{1}{2}\mathbf{r}^T A\mathbf{r} + \mathbf{r}^T \mathbf{d} + k I
+
+        where:
+
+        * :math:`A\in\mathbb{R}^{2N\times 2N}` is a symmetric matrix
+          representing the quadratic coefficients,
+        * :math:`\mathbf{d}\in\mathbb{R}^{2N}` is a real vector representing
+          the linear coefficients,
+        * :math:`k\in\mathbb{R}` represents the constant term, and
+        * :math:`\mathbf{r} = (\x_1,\dots,\x_N,\p_1,\dots,\p_N)` is the vector
+          of quadrature operators in :math:`xp`-ordering.
```

Comment:
Just curious, what would it be called if we used the ordering (x1,p1,x2,p2,...)?

Reply:
Good question, @nquesada? I just copied and pasted this from somewhere else in the SF codebase. I've seen one called the symmetric ordering, and one called the `xp`-ordering.